R/add_subprotocol.R

```diff
@@ -75,11 +68,7 @@ add_subprotocol <-
         "protocol code not in s*f-###-nl or s*f-###-en format"
       )
     }
-    assert_that(is.numeric(demote_header))
-    demote_choices <- c(0, 1, 2, -1)
-    if (!(demote_header %in% demote_choices)) {
-      stop(paste0("demote header must be one of ", demote_choices))
-    }
+    demote_choices <- match.arg(demote_choices)
```

Comment:
See previous comment. So I think you need to revert this or provide a working alternative for my original. (Note that the argument is named `demote_header` instead of `demote_choices`.)

Reply:
Good catch (wrong argument name). I'll have a further look to `match.arg()` to see whether this approach can be somehow maintained.

Reply:
Your test catched it :+1 (ran `R CMD check` locally): 

```
══ Failed tests ════════════════════════════════════════════════════════════════
── Error (test-add_subprotocol.R:33:3): Test that add subprotocol works ────────
Error: object 'demote_choices' not found
```

Reply:
Argument name fixed in 894fa97. A new suggestion added in 06c08b6, because I like the common approach for string arguments with multiple allowed values to show them as a vector in the usage section (docs) which can then be used by `match.arg()` inside the function to both check the inputted value and take the first as default if missing.

The same is mimicked  in 06c08b6 by putting the vector of allowed values inside the function definition and then calling these values with `formals()`.

I believe this makes it both easier to maintain (you don't need to repeat the defaults in the documentation) and it is still visible in the documentation (usage section) which values are allowed.

Is this OK for you @hansvancalster ?

Reply:
Note that I did not re-insert the precondition for a numeric input (BTW would better be `is.number()` = more strict). It already follows from `assert_that(demote_header %in% demote_choices, msg) `.

Reply:
Cool solution! All new stuff for me :-)